llama-index-core/llama_index/core/agent/runner/base.py

```diff
@@ -702,3 +729,134 @@ async def astream_chat(
     def undo_step(self, task_id: str) -> None:
         """Undo previous step."""
         raise NotImplementedError("undo_step not implemented")
+
+
+class BasePlanningAgentRunner(AgentRunner):
+    @abstractmethod
+    def create_plan(self, input: str, **kwargs: Any) -> str:
```

Comment:
Is there a dedicated data structure for "Plan" or is it just an id?

Reply:
I have a structure in my implementation, but I wanted to leave the base class more generic